src/System.Windows.Forms/src/System/Windows/Forms/FolderBrowserDialog.cs

```diff
@@ -228,10 +271,24 @@ private void SetDialogProperties(IFileDialog dialog)
 
             dialog.SetOptions(FOS.PICKFOLDERS | FOS.FORCEFILESYSTEM | FOS.FILEMUSTEXIST);
 
+            if (!string.IsNullOrEmpty(InitialDirectory))
```

Comment:
```suggestion
            if (!string.IsNullOrWhiteSpace(InitialDirectory))
```

Reply:
We appear to be inconsistent in our use - here we use the property, and below we use the field.
```suggestion
            if (!string.IsNullOrEmpty(_initialDirectory))
```

Reply:
> We appear to be inconsistent in our use

Leftover from FileDialog because I wanted to keep code that didn't need to differ identical. In FileDialog the getter does normalization to empty string and its mandatory to use the property, FolderBrowserDialog does it the other way round and normalizes in the setter.

I've updated the PR to keep the usage in the file consistent, instead of across files.

> IsNullOrWhiteSpace

If I do this change we start becoming inconsistent, some properties ignore blank strings, some don't. I copied this from FileDialog so FolderDialog being different from FileDialog is going to be weird.

Personally I don't like treating whitespace in blank strings as insignificant when you are doing it inconsistently (only in some places). Also it adds a special case instead of generalizing, in my opinion it would be much more sensible to treat leading/trailing whitespace to be insignificant rather than special casing a blank string. In this case you would be trimming in the setter, not checking IsNullOrWhiteSpace in every call site.

In the end its your choice, if you can confirm that you want IsNullOrWhiteSpace in this specific location and only there then I'll update, otherwise I think updating existing call sites belongs in a separate PR (including tests for changed behavior) and maybe an issue to do this conistently in the codebase (and perhaps discuss trimming).

Reply:
> Leftover from FileDialog because I wanted to keep code that didn't need to differ identical. In FileDialog the getter does normalization to empty string and its mandatory to use the property, FolderBrowserDialog does it the other way round and normalizes in the setter.

Yes, I noted this during the review. It is indeed different to how we generally do it.

Reply:
> > IsNullOrWhiteSpace
> 
> If I do this change we start becoming inconsistent, some properties ignore blank strings, some don't. I copied this from FileDialog so FolderDialog being different from FileDialog is going to be weird.

Aye, let's keep as is.